extensions/smallrye-fault-tolerance/deployment/src/main/java/io/quarkus/smallrye/faulttolerance/deployment/SmallRyeFaultToleranceProcessor.java

```diff
@@ -243,73 +236,79 @@ public void transform(TransformationContext ctx) {
     void validateFaultToleranceAnnotations(SmallRyeFaultToleranceRecorder recorder,
             ValidationPhaseBuildItem validationPhase,
             BeanArchiveIndexBuildItem beanArchiveIndexBuildItem,
+            AnnotationProxyBuildItem annotationProxy,
+            BuildProducer<GeneratedClassBuildItem> generatedClasses,
             BuildProducer<ValidationPhaseBuildItem.ValidationErrorBuildItem> errors) {
+
         AnnotationStore annotationStore = validationPhase.getContext().get(BuildExtension.Key.ANNOTATION_STORE);
-        Set<String> beanNames = new HashSet<>();
         IndexView index = beanArchiveIndexBuildItem.getIndex();
+        ClassOutput classOutput = new GeneratedClassGizmoAdaptor(generatedClasses, true); // TODO app class???
```

Comment:
@mkouba (or maybe @geoand) could you please advise how to determine a correct value for the 2nd argument here?

Apparently it should be `true` if the generated class is an "application class", but I have no idea what that means :-)

Reply:
IIRC, it should be true is this is expected to change to when the application is reloaded.

Reply:
@geoand is right, this boolean is used by the classloader during the dev mode. Application classes are "hot deployable". You can use a predicate built from the `io.quarkus.deployment.GeneratedClassGizmoAdaptor.isApplicationClass(String)`, that should work just fine (at least for classes that were not generated by a build step).